app/scenes/ProductForm.js

```diff
@@ -3,10 +3,17 @@ import { Actions } from 'react-native-router-flux';
 
 import ProductFormView from '../components/ProductFormView';
 import { requestCreateProduct } from '../reducers/products';
+import { categoriesToObject } from '../lib/helpers';
+import { requestCategories } from '../reducers/categories';
+
+const mapStateToProps = state => ({
+  categories: categoriesToObject(state.categories.list.records),
```

Comment:
No, mapping state is ok. What I'm concerning about is fetching categories each time we mounting different scenes.